sdks/python/container/base_image_requirements.txt

```diff
@@ -28,46 +28,46 @@ avro-python3==1.8.2;python_version>="3.4"
 fastavro==0.21.4
 crcmod==1.7
 dill==0.2.9
-future==0.16.0
+future==0.17.1
 futures==3.2.0;python_version<"3.0"
-grpcio==1.15.0
-hdfs==2.1.0
-httplib2==0.9.2
+grpcio==1.21.1
+hdfs==2.5.6
+httplib2==0.12.0
 mock==2.0.0
 oauth2client==3.0.0
-protobuf==3.6.1
-pyarrow==0.11.1
-pydot==1.2.4
-pyparsing==2.3.1
-pytz==2018.4
+protobuf==3.8.0
+pyarrow==0.13.0
+pydot==1.4.1
+pyparsing==2.4.0
```

Comment:
Do you know where this dependency comes from?

Reply:
Dependency chain is `pipeline_graph` -> `pydot` -> `dot_parser` -> `pyparsing`.

Reply:
We already have pydot in the list. And we do not list other transitive dependencies. Should we remove this from here?

Separately, it is actually a good idea to list all transitive dependencies. There was a mailing list discussion on this. I do not know if there is a JIRA issue about it.

Reply:
I removed `pyparsing` from the list.
I haven't found a ticket for listing all transitive dependencies. Could you please explain what it is? I will create a ticket when I have more details.

Reply:
Beam has a list of dependencies (listed in setup.py). These are Beam's direct dependency. Each of these direct dependency will have their own list of dependencies. These are transitive dependencies of Beam. Beam depends on them indirectly and in setup.py Beam does not list them.

In a worker environment it is beneficial to lock version of all Beam dependencies including the indirect ones.

Does it answer your question?

Reply:
Yes. Thanks for more details. I created a ticket https://issues.apache.org/jira/browse/BEAM-8084.